REF: Convert test_hashing to pytest idiom

gfyoung opened the pull request.

gfyoung committed:
REF: Convert test_hashing to pytest idiom

pep8speaks commented:
Hello @gfyoung! Thanks for submitting the PR.

 - In the file [`pandas/tests/util/test_hashing.py`](https://github.com/gfyoung/pandas/blob/f1bbe887646a373f1afa2cea877cc5e04fba4d60/pandas/tests/util/test_hashing.py), following                are the PEP8 issues :
> [Line 315:5](https://github.com/gfyoung/pandas/blob/f1bbe887646a373f1afa2cea877cc5e04fba4d60/pandas/tests/util/test_hashing.py#L315): [E741](https://duckduckgo.com/?q=pep8%20E741) ambiguous variable name 'l'

TomAugspurger commented on pandas/tests/util/test_hashing.py:
```diff
@@ -10,272 +10,320 @@
+
+    kwargs.pop("index", None)
+
+    a = hash_pandas_object(obj, **kwargs)
```
Hmm I know this is how it was before, but this looks buggy. We don't seem to ever call `_check_equal` with an `index=` kwarg, so we always use the default of `index=True`.

So I think we're testing that `hash_pandas_object(obj, index=True)` twice, and not checking that `hash_pandas_object(obj, index=False)` at all. Or am I missing something?

gfyoung replied:
Nope, you're not missing anything.  Let's fix that.

gfyoung committed:
BUG: Actually check index=True/False in some cases

_check_equal appeared to be checking index=True/False,
but we actually need to do explicitly pass in
index=True/False for this to work.

gfyoung commented:
@jreback @TomAugspurger : Addressed the comment above, and all is still green.  PTAL.

TomAugspurger commented:
LGTM. Instead of the index fixture you could also just call each of `index=False` and `index=True` in `_check_equal`, but either way works for me.

gfyoung commented:
> Instead of the index fixture you could also just call each of index=False and index=True in _check_equal, but either way works for me.

Thought about it.  Just wanted something a little more modular.
